### ISSUE TITLE
add communication layer between frontend and backend

### ISSUE THREAD

**Issue by GitMensch** [WebFreak001/code-debug#328]
> For different issues we need a communication between frontend and backend. 

_Originally posted by @GitMensch in https://github.com/WebFreak001/code-debug/issues/325#issuecomment-1060743744_

This is a show-stopper for at least #325, #326 as well as #132.
This is likely also needed for #318 #319 as all those are frontend related.

I do know that it is possible to send "CustomEvents" but I'm not sure if this would be enough for communication.

What seems possible is to open a local socket and send messages in both directions.

**Comment by WebFreak001** [comment]
See https://github.com/Microsoft/vscode/issues/49389

**Comment by GitMensch** [comment]
Yes, that sounds like the correct approach. Do you mind adding an example? A good case would actually be #326 - the code on the extension side is technically the two liner presented there.
That would be only the "backend to frontend" part, but that would be already very useful on its own.
As a second part for the other way around could be worked on with #325.
